AUDIO/AddLineToConversation.md

```diff
@@ -5,39 +5,31 @@ ns: AUDIO
 
 ```c
 // 0xC5EF963405593646 0x96CD0513
-void ADD_LINE_TO_CONVERSATION(int index, char* p1, char* p2, int p3, int p4, BOOL p5, BOOL p6, BOOL p7, BOOL p8, int p9, BOOL p10, BOOL p11, BOOL p12);
+void ADD_LINE_TO_CONVERSATION(int speakerConversationIndex, char* context, char* subtitle, int listenerNumber, int volumeType, BOOL isRandom, BOOL interruptible, BOOL ducksRadio, BOOL ducksScore, int audibility, BOOL headset, BOOL dontInterruptForSpecialAbility, BOOL isPadSpeakerRoute);
 ```
 
-```
-NOTE: ones that are -1, 0 - 35 are determined by a function where it gets a TextLabel from a global then runs,
-_GET_TEXT_SUBSTRING and depending on what the result is it goes in check order of 0 - 9 then A - Z then z (lowercase). So it will then return 0 - 35 or -1 if it's 'z'. The func to handle that ^^ is func_67 in dialog_handler.c atleast in TU27 Xbox360 scripts.
-p0 is -1, 0 - 35
-p1 is a char or string (whatever you wanna call it)
-p2 is Global 10597 + i * 6. 'i' is a while(i < 70) loop
-p3 is again -1, 0 - 35
-p4 is again -1, 0 - 35
-p5 is either 0 or 1 (bool ?)
-p6 is either 0 or 1 (The func to determine this is bool)
-p7 is either 0 or 1 (The func to determine this is bool)
-p8 is either 0 or 1 (The func to determine this is bool)
-p9 is 0 - 3 (Determined by func_60 in dialogue_handler.c)
-p10 is either 0 or 1 (The func to determine this is bool)
-p11 is either 0 or 1 (The func to determine this is bool)
-p12 is unknown as in TU27 X360 scripts it only goes to p11.
+```cpp
+enum eAudibility {
+	AUD_AUDIBILITY_NORMAL = 0,
+	AUD_AUDIBILITY_CLEAR = 1,
+	AUD_AUDIBILITY_CRITICAL = 2,
+	AUD_AUDIBILITY_LEAD_IN = 3
+}
```

Comment:
Instead of this, you should do table method since most of the new PR's for natives goes that way.

| Enum Name              | Value | Description          |
|------------------------|-------|----------------------|
| AUD_AUDIBILITY_NORMAL  | 0     | Normal audibility    |
| AUD_AUDIBILITY_CLEAR   | 1     | Clear audibility     |
| AUD_AUDIBILITY_CRITICAL| 2     | Critical audibility  |
| AUD_AUDIBILITY_LEAD_IN | 3     | Lead-in audibility   |

Reply:
I'll wait for @4mmonium's opinion on this, this is an enum ad listeningit as an enum would make it easier use & copy over a generic table.

Reply:
I prefer enumerations, for other things that are not enumerations, tables may be desired. Enumerations can be copied so they're ready to go in-code.